Убрать ворнинги линтера

Зашел я, например, в js календаря, а он весь красный)

код почти тот(не совсем календарь), но я не могу убрать ворнинги твоего линтера, у меня сейчас 19 ошибок и про карту там одна. 

> 
D:\test\blog-markup\src\components\location\index.js
  22:11  error  'marker' is assigned a value but never used  no-unused-vars

при этом правила он нашел 

> (node:7848) [ESLINT_LEGACY_OBJECT_REST_SPREAD] DeprecationWarning: The 'parserOp                      tions.ecmaFeatures.experimentalObjectRestSpread' option is deprecated. Use 'pars                      erOptions.ecmaVersion' instead. (found in "**airbnb-base**")

что я делаю не так?

--- a/src/components/arrow-button/arrow-button.styl
+++ b/src/components/arrow-button/arrow-button.styl
@@ -19,29 +19,29 @@
     transform rotateZ(180deg)
     
   &__border
-      stroke #4eb7a8
-      fill transparent
+    stroke #4eb7a8
+    fill transparent
       
   &__pointer
-      stroke #4eb7a8
+    stroke #4eb7a8
       
   &:hover
-     .arrow-button__border
+    & ^[0]__border
       stroke #4eb7a8
       fill #4eb7a8
       
-     .arrow-button__pointer
+    & ^[0]__pointer
       stroke white
       
   &--disabled,
   &--disabled:hover
     cursor not-allowed
     
-    .arrow-button__border
+    & ^[0]__border
       stroke grey
       fill grey
       
-    .arrow-button__pointer
+    & ^[0]__pointer
       stroke white
 
 

--- a/src/components/button/button.js
+++ b/src/components/button/button.js
@@ -1,4 +1,4 @@
-import Ripple from './ripple.js';
+import Ripple from './ripple';
 
 const buttons = document.querySelectorAll('.js-ripple');
-buttons.forEach( button => new Ripple(button));
+buttons.forEach(button => new Ripple(button));

--- a/src/components/button/button.styl
+++ b/src/components/button/button.styl
@@ -20,6 +20,7 @@ text-red = #e34a26
   letter-spacing .5px
   font-size 12px
   font-weight bold
+  outline none
   text-transform uppercase
   position relative
   border 1px solid var(--thecolor)
@@ -48,11 +49,11 @@ text-red = #e34a26
     color white !important
     width 100%
     
- &--small-padding 
+  &--small-padding 
     width 100%
     padding-left 10px
     padding-right 10px
-    
+      
   &--error
     --thecolor : medium-red
     color text-red

--- a/src/components/button/ripple.js
+++ b/src/components/button/ripple.js
@@ -1,35 +1,35 @@
 class Ripple {
-  constructor(element, {baseBlock="ripple"}={}) {
+  constructor(element, { baseBlock = 'ripple' }={}) {
     this.element = element instanceof HTMLElement ? element : document.querySelector(element);
     if (!this.element) throw new Error(`There is no node corresponding to ${element} in this document`);
     this.baseBlock = baseBlock;
     this.rippleContainer = this.createElement("div", `${this.baseBlock}__container`);
     this.element.appendChild(this.rippleContainer);
     this.element.addEventListener('click', this.addRipple.bind(this));
-  };
+  }
 
   addRipple (event) {
-    let container = this.rippleContainer;
-    let rippler = this.createElement('span', `${this.baseBlock}__rippler`);
-    let size = container.offsetWidth;
-    let pos = container.getBoundingClientRect();
-    let x = event.pageX - pos.left - (size / 2) - window.pageXOffset;
-    let y = event.pageY - pos.top - (size / 2) - window.pageYOffset;
-    let style = `top:${y}px; 
+    const container = this.rippleContainer;
+    const rippler = this.createElement('span', `${this.baseBlock}__rippler`);
+    const size = container.offsetWidth;
+    const pos = container.getBoundingClientRect();
+    const x = event.pageX - pos.left - (size / 2) - window.pageXOffset;
+    const y = event.pageY - pos.top - (size / 2) - window.pageYOffset;
+    const style = `top:${y}px;
                  left:${x}px; 
                  height:${size}px; 
                  width: ${size}px;
                  `;
-    rippler.style=style;
+    rippler.style = style;
     this.rippleContainer.appendChild(rippler);
     window.setTimeout(() => rippler.remove(), 2000);
   }
 
   createElement(tag, className) {
-    let el = document.createElement(tag);
+    const el = document.createElement(tag);
     el.classList.add(className);
     return el;
   }
-};
+}
 
 export default Ripple;

--- a/src/components/calendar/calendar.styl
+++ b/src/components/calendar/calendar.styl
@@ -33,81 +33,81 @@
     font-weight: bold;
     letter-spacing: 1.1px;
 
-.ui-datepicker-inline
-  width 100%
-  
-.ui-datepicker-header
-  background-color #d54c2c
-  position relative
-  height 42px
-  padding-top 17px
-  font-weight 300
-  font-size 24px
-  color white
-  line-height 0
-  
-.ui-datepicker-title
-  text-align center
-  padding-top: 4px;
-
-.ui-datepicker-prev
-.ui-datepicker-next
-  position absolute
-  cursor pointer
-  top 10px
-  &:hover svg path
-    stroke var(--support-color)
-  
-.ui-datepicker-prev
-  left 6px  
-  height 18px
-  width 18px
-  
-.ui-datepicker-next
-  right 8px
-  top 12px
-  height 18px
-  width 18px
-  
-.ui-datepicker-year
-  display none
-  
-.ui-datepicker-calendar
-  border-collapse collapse
-  background-color #f2f2f2
-  
-  th
-    background-color var(--main-color)
+  .ui-datepicker-inline
+    width 100%
+    
+  .ui-datepicker-header
+    background-color #d54c2c
+    position relative
+    height 42px
+    padding-top 17px
+    font-weight 300
+    font-size 24px
     color white
-    text-transform uppercase
-    font-size .6rem
-    padding-top: 9px;
-    padding-bottom: 6px;
+    line-height 0
     
-  td
-    width: 40px
-    height 40px
-    border 1px solid white
+  .ui-datepicker-title
     text-align center
-    vertical-align middle
+    padding-top: 4px;
+
+  .ui-datepicker-prev
+  .ui-datepicker-next
+    position absolute
+    cursor pointer
+    top 10px
+    &:hover svg path
+      stroke var(--support-color)
     
-    &.ui-datepicker-other-month
+  .ui-datepicker-prev
+    left 6px  
+    height 18px
+    width 18px
+    
+  .ui-datepicker-next
+    right 8px
+    top 12px
+    height 18px
+    width 18px
+    
+  .ui-datepicker-year
+    display none
+    
+  .ui-datepicker-calendar
+    border-collapse collapse
+    background-color #f2f2f2
+    
+    th
+      background-color var(--main-color)
       color white
+      text-transform uppercase
+      font-size .6rem
+      padding-top: 9px;
+      padding-bottom: 6px;
       
-    a
-      display flex
-      justify-content center
-      align-items center
-      width 100%
-      height 100%
-      text-decoration none
-      color var(--text-light)
+    td
+      width: 40px
+      height 40px
+      border 1px solid white
+      text-align center
+      vertical-align middle
       
-      &.ui-state-active
-        background-color var(--main-color)
+      &.ui-datepicker-other-month
         color white
         
-      &:hover
-        background-color var(--support-color)
-        color white
-      
+      a
+        display flex
+        justify-content center
+        align-items center
+        width 100%
+        height 100%
+        text-decoration none
+        color var(--text-light)
+        
+        &.ui-state-active
+          background-color var(--main-color)
+          color white
+          
+        &:hover
+          background-color var(--support-color)
+          color white
+        

--- a/src/components/calendar/index.js
+++ b/src/components/calendar/index.js
@@ -1,38 +1,38 @@
-import $ from "jquery"
+/* eslint prefer-destructuring:0 */
+import $ from 'jquery';
 import datepickerFactory from 'jquery-datepicker';
-
 
 class Calendar {
   constructor(element) {
     datepickerFactory($);
-    this.currentDate = null
-    this.head = $(element).find(".js-calendar__head")
-    this.today = $(element).find(".js-calendar__today")
+    this.currentDate = null;
+    this.head = $(element).find('.js-calendar__head');
+    this.today = $(element).find('.js-calendar__today');
     this.calendar = $(element).datepicker({
-      prevText: '<svg  style="width:100%; height:100%" viewBox="0 0 30 30" transform="rotate(180.1)">' + $(".arrow-icon").html() + "</svg>",
-      nextText: '<svg style="width:100%; height:100%" viewBox="0 0 30 30">' + $(".arrow-icon").html() + "</svg>",
+      prevText: `<svg  style="width:100%; height:100%" viewBox="0 0 30 30" transform="rotate(180.1)"> ${$('.arrow-icon').html()}</svg>`,
+      nextText: `<svg style="width:100%; height:100%" viewBox="0 0 30 30"> ${$('.arrow-icon').html()}</svg>`,
       firstDay: 1,
       showOtherMonths: true,
-      dayNamesMin: [ "sun", "mon", "tue", "wed", "thu", "fri", "sat" ],
-      onSelect: this.setHeader.bind(this)
+      dayNamesMin: ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'],
+      onSelect: this.setHeader.bind(this),
     });
-    this.calendar.prepend(this.head)
-    this.calendar.append(this.today)
-    this.setTodayDate()
-    this.today.click(this.setTodayDate.bind(this))
+    this.calendar.prepend(this.head);
+    this.calendar.append(this.today);
+    this.setTodayDate();
+    this.today.click(this.setTodayDate.bind(this));
   }
 
   setTodayDate() {
-    this.calendar.datepicker( "setDate", new Date() )
-    let date = this.calendar.datepicker( "getDate" );
-    this.head.html(date.getDate())
+    this.calendar.datepicker('setDate', new Date());
+    const date = this.calendar.datepicker('getDate');
+    this.head.html(date.getDate());
   }
 
   setHeader(date) { // date === "mm/dd/yyyy"
-    [ ,this.currentDate, ] = date.split('/')
-    this.head.html(this.currentDate)
+    this.currentDate = date.split('/')[1];
+    this.head.html(this.currentDate);
   }
 }
 
-$('.js-calendar').each((index, element) => new Calendar(element))
+$('.js-calendar').each((index, element) => new Calendar(element));
 

--- a/src/components/donut/index.js
+++ b/src/components/donut/index.js
@@ -1,26 +1,27 @@
-import $ from "jquery";
+import $ from 'jquery';
 import createSVGDoughnut from 'create-svg-doughnut';
 
 class Donut {
   constructor(element) {
+    this.node = document.createElementNS("http://www.w3.org/2000/svg", 'text');
     const segments = $(element).data('chart-segments');
     const innerRadius = +$(element).data('chart-inner-radius');
     const outerRadius = +$(element).data('chart-outer-radius');
-    const vals = segments.map( seg => +seg[0] )
-    const pals = segments.map( seg => seg[1] )
+    const vals = segments.map(seg => +seg[0]);
+    const pals = segments.map(seg => seg[1]);
     const doughnut = createSVGDoughnut(vals, outerRadius, innerRadius, pals);
-    element.appendChild(doughnut)
+    element.appendChild(doughnut);
   }
 
-  createSvgNode(nodeName, nodeAttributes) {
-    let node = document.createElementNS("http://www.w3.org/2000/svg", nodeName);
-    Object.keys(nodeAttributes).forEach(attribute => node.setAttributeNS(null, attribute, nodeAttributes[attribute]));
-    return node;
+  createSvgNode(nodeAttributes) {
+    Object.keys(nodeAttributes)
+      .forEach(attribute => this.node.setAttributeNS(null, attribute, nodeAttributes[attribute]));
+    return this.node;
   }
 }
 
 $(document).ready(() => {
-  $(".donut").each((i, element) => new Donut(element));
-})
+  $('.donut').each((i, element) => new Donut(element));
+});
 
 

--- a/src/components/event/event.pug
+++ b/src/components/event/event.pug
@@ -1,7 +1,7 @@
 include ../button/button.pug
 
 mixin event({image, date:{day, month}, title, copy="lorem ipsum"}={})
-    .event(style="--image-location: url(" + image + ");")
+    .event(style=`--image-location: url("${image}");`)
         .event__image
         .event__body
             .event__date
@@ -10,4 +10,4 @@ mixin event({image, date:{day, month}, title, copy="lorem ipsum"}={})
             h2.event__title=title
             .event__copy=copy
             .event__button-back
-                +button("Add to calendar", {size:'small', theme: "small-padding"})
+                +button("Add to calendar", {size: 'small'})

--- a/src/components/input/input.pug
+++ b/src/components/input/input.pug
@@ -1,5 +1,4 @@
 mixin input(placeholder, {pattern, errMessage="Error", sucMessage="Thanks!"}={})
- 
     .input
         .input__back
             input(type='text', placeholder=placeholder, pattern=pattern).input__actual

--- a/src/components/input/input.styl
+++ b/src/components/input/input.styl
@@ -15,11 +15,11 @@
     border-radius 4px
     border none
     
-    &:focus:invalid~.input__message--error
+    &:focus:invalid~^[0]__message--error
       display inline-block
       --state-color var(--main-color)
       
-    &:focus:valid~.input__message--success
+    &:focus:valid~^[0]__message--success
       display inline-block
       --state-color var(--support-color)
       

--- a/src/components/location/index.js
+++ b/src/components/location/index.js
@@ -31,10 +31,7 @@ class Location {
       if (status === google.maps.GeocoderStatus.OK) {
         let [res1, res2] = results; 
         let result = (results.length > 1) ? res1 : res2;
-        $(".js-location__address1").html(result.address_components[0].long_name + ", ")
-        $(".js-location__address2").html(result.address_components[1].long_name + ' ')
-        $(".js-location__address3").html(result.address_components[2].long_name + ', ')
-        $(".js-location__address4").html(result.address_components[3].short_name + ' ')
+        $(".js-location__address").html(result.formatted_address)
       }  
     });    
   }

--- a/src/components/location/index.js
+++ b/src/components/location/index.js
@@ -1,41 +1,43 @@
-import $ from "jquery"
-import maps from "google-maps"
+/* eslint no-new: "off" */
+import $ from 'jquery';
+import maps from 'google-maps';
 
 class Location {
   constructor(element) {
-    this.el = element
-    this.lat = $(this.el).data("lat")
-    this.lng = $(this.el).data("lng")
-    maps.KEY = "AIzaSyDZZlCXwgC_cO0bD_3nsEFkyD_Gf3PbG5w"
-    maps.LANGUAGE = "en" 
+    this.el = element;
+    this.lat = $(this.el).data('lat');
+    this.lng = $(this.el).data('lng');
+    maps.KEY = 'AIzaSyDZZlCXwgC_cO0bD_3nsEFkyD_Gf3PbG5w';
+    maps.LANGUAGE = 'en';
     maps.load(this.createMap.bind(this));
   }
 
   createMap(google) {
-    var map = new google.maps.Map(this.el, { // eslint-disable-line no-new
-        center: {lat: this.lat, lng: this.lng},
-        zoom: 12,
-        disableDefaultUI: true
-      });
-    var geocoder = new google.maps.Geocoder();
-    var geolocate = new google.maps.LatLng(this.lat, this.lng)
-    var marker = new google.maps.Marker({
-      position: {lat: this.lat, lng: this.lng},
-      map: map,
+    const map = new google.maps.Map(this.el, {
+      center: { lat: this.lat, lng: this.lng },
+      zoom: 12,
+      disableDefaultUI: true,
+    });
+    const geocoder = new google.maps.Geocoder();
+    const geolocate = new google.maps.LatLng(this.lat, this.lng);
+    new google.maps.Marker({
+      position: { lat: this.lat, lng: this.lng },
+      map,
       icon: '../images/Marker.png',
-      scale: .2,
-      title: 'Hello World!'
+      scale: 0.2,
+      title: 'Hello World!',
     });
 
-    geocoder.geocode({'latLng': geolocate}, function(results, status) {
+    geocoder.geocode({ latLng: geolocate }, (results, status) => {
       if (status === google.maps.GeocoderStatus.OK) {
-        let [res1, res2] = results; 
-        let result = (results.length > 1) ? res1 : res2;
-        $(".js-location__address").html(result.formatted_address)
-      }  
-    });    
+        const [res1, res2] = results;
+        const result = (results.length > 1) ? res1 : res2;
+        $('.js-location__address').html(result.formatted_address);
+      }
+    });
   }
 }
 
-$('.js-location').each((index, element) => new Location(element))
+$('.js-location').each((index, element) => new Location(element));
+
 

--- a/src/components/location/location.pug
+++ b/src/components/location/location.pug
@@ -4,12 +4,8 @@ mixin location({ lat=.28, lng=110.24 }={})
         .location__bar
             .location__message Meet us!
             .location__address
-                .location__address-line-1
-                    span.js-location__address1
-                    span.js-location__address2
-                .location__address-line-2
-                    span.js-location__address3
-                    span.js-location__address4
+                .location__address-line
+                    span.js-location__address
             .location__controls
                 .location__icon
                     include /src/components/location/target-icon.svg

--- a/src/components/location/location.styl
+++ b/src/components/location/location.styl
@@ -37,7 +37,6 @@
     text-transform uppercase
     font-size 12px
   
-
 .target-icon__group
   stroke #fff
   fill transparent

--- a/src/components/messaging/messaging.styl
+++ b/src/components/messaging/messaging.styl
@@ -6,9 +6,11 @@ pads = 23px
   width 288px
   border-radius 4px
   overflow hidden
+  
   &&--inverse
     --color2 : var(--main-color)
     --color1 : var(--support-color)
+    
   &__title
     line-height 100%
     font-size 34px
@@ -16,15 +18,12 @@ pads = 23px
     font-weight 300
     height 118px
     background-color var(--color2)
-      
-      
 
   &__title-text
     padding-top 22px
     letter-spacing -1.5px
     text-align center
     
-      
   &__icons
     width 100%
     height 34px
@@ -51,6 +50,7 @@ pads = 23px
     overflow hidden
     background-color var(--dark-grey)
     border 5px solid var(--light-grey)
+    
   &__incoming-message
     padding 15px 20px 22px
     background-color white
@@ -60,8 +60,10 @@ pads = 23px
     margin-top 15px
     color var(--text-dark)
     position relative
+    
   &__incoming-message-text
     letter-spacing 1px
+    
   &__outgoing-message
     height 90px 
     border 3px solid white
@@ -73,10 +75,13 @@ pads = 23px
   &__body
     padding 48px pads 26px
     background-color #f2f2f2
+    
   &__form 
-    margin-top 17px
+    margin-top 
+    
   &__button
     margin-top 15px
+    
   &__bubble-tail
     fill white
     position absolute

--- a/src/components/news/news.pug
+++ b/src/components/news/news.pug
@@ -1,5 +1,5 @@
 mixin news({image, title, date = (new Date()).toLocaleDateString(), copy="lorem ipsum"}={})
-    .news(style="--image-location: url(" + image + ");")
+    .news(style=`--image-location: url("${image}");`)
         .news__image
         .news__body
             h2.news__title=title

--- a/src/components/percentage/index.js
+++ b/src/components/percentage/index.js
@@ -1,36 +1,37 @@
-import $ from "jquery";
+import $ from 'jquery';
 import createSVGDoughnut from 'create-svg-doughnut';
 
 class Percentage {
   constructor(element) {
     this.element = element;
+    this.node = document.createElementNS('http://www.w3.org/2000/svg', 'text');
     const segments = $(element).data('chart-segments');
     const innerRadius = +$(element).data('chart-inner-radius');
     const outerRadius = +$(element).data('chart-outer-radius');
-    const vals = segments.map( seg => +seg[0] )
-    const pals = segments.map( seg => seg[1] )
+    const vals = segments.map(seg => +seg[0]);
+    const pals = segments.map(seg => seg[1]);
     this.doughnut = createSVGDoughnut(vals, outerRadius, innerRadius, pals);
-    this.addText()
-    element.appendChild(this.doughnut)
+    this.addText();
+    element.appendChild(this.doughnut);
   }
 
   addText() {
     const chartText = $(this.element).data('chart-text').toString() || '';
-    const svgText = this.createSvgNode('text', { x: 47, y: 60, class: "percentage__text-in-circle", 'text-anchor':"middle" });
+    const svgText = this.createSvgNode({
+      x: 47, y: 60, class: 'percentage__text-in-circle', 'text-anchor': 'middle',
+    });
     const textNode = document.createTextNode(chartText);
     svgText.appendChild(textNode);
     this.doughnut.appendChild(svgText);
   }
 
-  createSvgNode(nodeName, nodeAttributes) {
-    let node = document.createElementNS("http://www.w3.org/2000/svg", nodeName);
-    Object.keys(nodeAttributes).forEach(attribute => node.setAttributeNS(null, attribute, nodeAttributes[attribute]));
-    return node;
+  createSvgNode(nodeAttributes) {
+    Object.keys(nodeAttributes)
+      .forEach(attribute => this.node.setAttributeNS(null, attribute, nodeAttributes[attribute]));
+    return this.node;
   }
 }
 
 $(document).ready(() => {
-  $(".percentage").each((i, element) => new Percentage(element));
-})
-
-
+  $('.percentage').each((i, element) => new Percentage(element));
+});

--- a/src/components/profile/profile.styl
+++ b/src/components/profile/profile.styl
@@ -21,11 +21,13 @@
     position relative
     margin-left .75rem
     color #4f4f4f
+    
   &__bubble-tail
     position absolute
     top calc(50% - .75rem)
     left -.5625rem
     fill var(--light-grey)
+    
   &__name
     font-size 1.125rem
     &:hover
@@ -49,13 +51,13 @@
     &:hover
       fill var(--main-color) 
       
-.profile__head:hover + .profile__story
-  background-color var(--support-color)
-  color white
-  .profile__bubble-tail
-    fill var(--support-color)
-  .icon
-    fill white
+  &__head:hover + &__story
+    background-color var(--support-color)
+    color white
+    
+    & ^[0]__bubble-tail
+      fill var(--support-color)
+
     
     
    

--- a/src/components/slider/index.js
+++ b/src/components/slider/index.js
@@ -1,33 +1,34 @@
-import $ from "jquery"
-import noUiSlider from "nouislider"
+import $ from 'jquery';
+import noUiSlider from 'nouislider';
 
 class Slider {
   constructor(element) {
+    this.element = element;
     const maxValue = $(element).data('max-value');
     const parts = $(element).data('parts');
     noUiSlider.create(element, {
-        range: {
-          min: 0,
-          max: maxValue
-        },
-        tooltips: [ !parts ],
-        start: [ Math.floor(maxValue / 2) ],
-        step: parts ? Math.floor(maxValue / (parts - 1)) : 1,
-        format: {
-          to: value => value | 0,
-          from: value => value | 0
-        },
-        pips: parts > 2 ? { mode: 'count', values: parts  } : undefined
-    })
+      range: {
+        min: 0,
+        max: maxValue,
+      },
+      tooltips: [ !parts ],
+      start: [Math.floor(maxValue / 2)],
+      step: parts ? Math.floor(maxValue / (parts - 1)) : 1,
+      format: {
+        to: value => Math.floor(value),
+        from: value => Math.floor(value),
+      },
+      pips: parts > 2 ? { mode: 'count', values: parts } : undefined,
+    });
     if (parts) {
-      var pips = $(element).find('.noUi-value');
-      pips.each((index, pip) => $(pip).click((event) => this.clickOnPip(element, event)))
+      const pips = $(element).find('.noUi-value');
+      pips.each((index, pip) => $(pip).click(event => this.clickOnPip(element, event)));
     }
   }
 
-  clickOnPip (el, event) {
-    var value = Number(event.currentTarget.getAttribute('data-value'));
-    el.noUiSlider.set(value);
+  clickOnPip(el, event) {
+    const value = Number(event.currentTarget.getAttribute('data-value'));
+    this.element.noUiSlider.set(value);
   }
 }
 

--- a/src/components/slider/slider.pug
+++ b/src/components/slider/slider.pug
@@ -3,5 +3,5 @@ mixin slider(maxValue, {parts="", pips="", color="1"}={})
       data-max-value=maxValue,
       data-parts= parts,
       data-pips-one= pips,
-      class= color == "1" ? "slider--color--one" : "slider--color--two"
+      class= color == "1" ? "slider--color--orange" : "slider--color--green"
     )

--- a/src/components/slider/slider.styl
+++ b/src/components/slider/slider.styl
@@ -6,80 +6,77 @@
     --slider-color : var(--support-color)
   &--color--orange
     --slider-color : var(--main-color)
-  *
-    &:active,
-    &:hover, 
-    &:focus
-      outline: 0
-      outline-offset: -2
+
   &[data-parts]
     margin-bottom 0px  
-      
-.noUi-target
-  border none
-  box-shadow none
-  
-.noUi-horizontal
-  height 4px  
-  
-  .noUi-tooltip 
-    display: block;
+        
+  &.noUi-target
     border none
-    outline none
-    color white
-    font-size 13px
-    border-radius 5px
-    padding 6px 10px 5px
-    line-height 100%
-    bottom: 141%
-    background-color var(--slider-color)
-    
-    &:after
-      background-color transparent  
-      content ""
-      position absolute
-      top 92%
-      width: 0px
-      height 0px
-      border-radius 4px
-      border-left 6px solid transparent
-      border-right 6px solid transparent
-      border-top 7px solid var(--slider-color) 
-      
-  .noUi-handle
-    width 20px
-    height 20px
-    top -3px
-    border-radius 50%
     box-shadow none
-    border none
-    background-color var(--slider-color)
     
-    &:before,
-    &:after
+  &.noUi-horizontal
+    height 4px 
+    
+    .noUi-tooltip 
+      display: block;
+      border none
+      outline none
+      color white
+      font-size 13px
+      border-radius 5px
+      padding 6px 10px 5px
+      line-height 100%
+      bottom: 141%
+      background-color var(--slider-color)
+      
+      &:after
+        background-color transparent  
+        content ""
+        position absolute
+        top 92%
+        left 33%
+        width 0px
+        height 0px
+        border-radius 4px
+        border-left 6px solid transparent
+        border-right 6px solid transparent
+        border-top 7px solid var(--slider-color) 
+        
+    .noUi-handle
+      width 20px
+      height 20px
+      top -7px
+      right -8px !important
+      border-radius 50%
+      box-shadow none
+      border none
+      background-color var(--slider-color)
+      outline none
+      
+      &:before,
+      &:after
+        display none
+
+  .noUi-pips-horizontal
+    padding .26rem 0 0 
+    font-size .70rem
+    color #cccccc
+    font-weight bold
+
+    .noUi-marker
       display none
 
-.noUi-pips-horizontal
-  padding .26rem 0 0 
-  font-size .70rem
-  color #cccccc
-  font-weight bold
+    .noUi-value
+      cursor pointer
+      
+    .noUi-connect, .noUi-origin
+      top -5px
 
-.noUi-marker
-  display none
+    .noUi-handle .noUi-tooltip 
+      display: flex
+      justify-content center
+      
 
-.noUi-value
-  cursor pointer
-  
-.noUi-connect, .noUi-origin
-  top -5px
-
-.noUi-handle .noUi-tooltip 
-  display: flex
-  justify-content center
-  
-html:not([dir=rtl]) .noUi-horizontal .noUi-handle
-  right -8px
 
       
   

--- a/src/components/slider/slider.styl
+++ b/src/components/slider/slider.styl
@@ -2,10 +2,10 @@
 
 .slider
   background-color var(--slider-color)
-  &--color--one
-    --slider-color : var(--main-color)
-  &--color--two
+  &--color--green
     --slider-color : var(--support-color)
+  &--color--orange
+    --slider-color : var(--main-color)
   *
     &:active,
     &:hover, 

--- a/src/components/tickbox/tickbox.styl
+++ b/src/components/tickbox/tickbox.styl
@@ -17,7 +17,7 @@
   &__input
     display none
     
-    &:checked + .tickbox__label
+    &:checked + ^[0]__label
       background-color var(--support-color)
       
       .tickbox__mark

--- a/src/components/toggle/toggle.styl
+++ b/src/components/toggle/toggle.styl
@@ -34,14 +34,14 @@
     border-radius 50%
     background-color white
 
-  .toggle__actual:checked
-    &+.toggle__label 
+  &__actual:checked
+    &+^[0]__label 
       background-color var(--support-color)
       
-      & > .toggle__circle
+      & > ^[0]__circle
         order 2  
         
-    &+.toggle__label:after
+    &+^[0]__label:after
       content "on" 
       order 1
 

--- a/src/components/video/index.js
+++ b/src/components/video/index.js
@@ -1,52 +1,51 @@
+/* eslint comma-dangle:0 */
 import $ from 'jquery'
 
 class Video {
   constructor(element) {
     this.$mediaPlayer = $(element).find('.js-video__actual');
-    this.mediaPlayer = this.$mediaPlayer[0]
+    this.mediaPlayer = this.$mediaPlayer[0];
     this.progressBar = $(element).find('.js-video__progress-bar');
     this.progressPosition = $(element).find('.js-video__progress-position');
     this.fullscreenButton = $(element).find('.js-video__fullscreen');
-    this.playButton = $(element).find(".js-video__play-pause-button")
-    this.icons = $(element).find(".js-video__pp-icon")
+    this.playButton = $(element).find('.js-video__play-pause-button');
+    this.icons = $(element).find('.js-video__pp-icon');
     this.mediaPlayer.controls = false;
-    
-    this.$mediaPlayer.on('timeupdate', this.updateProgressBar(this.mediaPlayer, this.progressPosition).bind(this))
-    this.playButton.click(this.togglePlayPause(this.icons, this.mediaPlayer).bind(this))
-    this.progressBar.click(this.setVideoTime(this.mediaPlayer, this.progressBar, this.progressPosition).bind(this))
-    this.fullscreenButton.click(() => this.setFullscreen(this.mediaPlayer).bind(this))
+
+    this.$mediaPlayer.on('timeupdate', this.updateProgressBar.bind(this));
+    this.playButton.click(this.togglePlayPause.bind(this));
+    this.progressBar.click(
+      this.setVideoTime.bind(this)
+    );
+    this.fullscreenButton.click(this.setFullscreen.bind(this));
   }
 
-  updateProgressBar(mediaPlayer, progressPosition) {
-    return function handler() {
-      const percentage = Math.floor((100 / mediaPlayer.duration) * mediaPlayer.currentTime);
-      progressPosition.css({width: percentage + '%'});
-    }
+  updateProgressBar() {
+    const percentage = Math.floor(
+      (100 / this.mediaPlayer.duration) * this.mediaPlayer.currentTime
+    );
+    this.progressPosition.css({ width: `${percentage}%` });
   }
 
-  setVideoTime(mediaPlayer, progressBar, progressPosition) {
-    return function handler(evt) {
-      const ratio = evt.offsetX / progressBar.width()
-      const mPlayer = mediaPlayer;
-      mPlayer.currentTime = ratio * mediaPlayer.duration
-      this.updateProgressBar(mediaPlayer, progressPosition)()
-    }
+  setVideoTime(evt) {
+    const ratio = evt.offsetX / this.progressBar.width();
+    this.mediaPlayer.currentTime = ratio * this.mediaPlayer.duration;
+    this.updateProgressBar.bind(this);
   }
-  
-  togglePlayPause(icons, mediaPlayer) {
-    return function handler() {
-      if (mediaPlayer.paused || mediaPlayer.ended) {
-        mediaPlayer
+
+  togglePlayPause() {
+    if (this.mediaPlayer.paused || this.mediaPlayer.ended) {
+      this.mediaPlayer
         .play()
-        .then(() => icons.toggleClass('video__pause-icon--hidden '));
-      } else {
-        mediaPlayer.pause()
-        icons.toggleClass('video__pause-icon--hidden ');
-      }
+        .then(() => this.icons.toggleClass('video__pause-icon--hidden '));
+    } else {
+      this.mediaPlayer.pause();
+      this.icons.toggleClass('video__pause-icon--hidden ');
     }
   }
 
-  setFullscreen(elem) {
+  setFullscreen() {
+    const elem = this.mediaPlayer;
     if (elem.requestFullscreen) {
       elem.requestFullscreen();
     } else if (elem.mozRequestFullScreen) {
@@ -57,4 +56,4 @@ class Video {
   }
 }
 
-$('.js-video').each( (ind, element) => new Video(element) )
+$('.js-video').each((ind, element) => new Video(element));

--- a/src/demo/app.js
+++ b/src/demo/app.js
@@ -1,5 +1,5 @@
-import '../components/components'
-import './site.blocks/photo-grid/photo-grid.js'
+import '../components/components';
+import './site.blocks/photo-grid/photo-grid';
 
 
 

--- a/src/demo/site.blocks/photo-grid/photo-grid.js
+++ b/src/demo/site.blocks/photo-grid/photo-grid.js
@@ -1,41 +1,41 @@
-import $ from 'jquery'
+import $ from 'jquery';
 
 class PhotoGrid {
   constructor(element) {
-    this.prevTarget = null
-    this.imageList = $(element).find(".js-photo-grid__item")
-    this.bigImage =$(element).find(".js-photo-grid__image-actual")  
-    this.leftControl = $(".js-photo-grid__control--left")
-    this.rightControl = $(".js-photo-grid__control--right")
+    this.prevTarget = null;
+    this.imageList = $(element).find('.js-photo-grid__item');
+    this.bigImage = $(element).find('.js-photo-grid__image-actual');
+    this.leftControl = $('.js-photo-grid__control--left');
+    this.rightControl = $('.js-photo-grid__control--right');
 
-    this.imageList.click(this.loadImage.bind(this))
-    this.leftControl.click(this.stepLeft.bind(this))
-    this.rightControl.click(this.stepRight.bind(this))
+    this.imageList.click(this.loadImage.bind(this));
+    this.leftControl.click(this.stepLeft.bind(this));
+    this.rightControl.click(this.stepRight.bind(this));
 
-    this.activate( $(this.imageList[0]) )
+    this.activate($(this.imageList[0]));
   }
 
   loadImage(evt) {
-    this.activate($(evt.currentTarget))
+    this.activate($(evt.currentTarget));
   }
 
-  stepRight(evt) {
-    this.activate( this.prevTarget.next().length ? this.prevTarget.next() : this.imageList.first() )
+  stepRight() {
+    this.activate(this.prevTarget.next().length ? this.prevTarget.next() : this.imageList.first());
   }
 
-  stepLeft(evt) {
-    this.activate( this.prevTarget.prev().length ? this.prevTarget.prev() : this.imageList.last() )
+  stepLeft() {
+    this.activate(this.prevTarget.prev().length ? this.prevTarget.prev() : this.imageList.last());
   }
 
   activate(target) {
-    this.bigImage.attr('src', target.data("image"))
-    if (this.prevTarget) this.prevTarget.removeClass("active") 
-    target.addClass("active")
-    this.prevTarget = target
-  } 
+    this.bigImage.attr('src', target.data('image'));
+    if (this.prevTarget) this.prevTarget.removeClass('active');
+    target.addClass('active');
+    this.prevTarget = target;
+  }
 }
 
-$('.js-photo-grid').each((index, element) => new PhotoGrid(element))
+$('.js-photo-grid').each((index, element) => new PhotoGrid(element));
 
 
 

